src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,49 @@
+package agh.ics.oop;
+
+public class RectangularMap implements IWorldMap {
+
+    private final Animal[][] map;
+
+    public RectangularMap(int width, int height) {
+        this.map = new Animal[width][height];
+    }
+
+    public boolean canMoveTo(Vector2d position) {
+        return position.x>=0
+                && position.x<this.map.length
+                && position.y>=0
+                && position.y<this.map[0].length
+                && this.map[position.x][position.y] == null;
+    }
+
+    public boolean place(Animal animal) {
+        Vector2d animalPos = animal.getPosition();
+        if (!isOccupied(new Vector2d(animalPos.x, animalPos.y))) {
+            this.map[animalPos.x][animalPos.y] = animal;
+            return true;
+        }
+        return false;
+    }
+
+    public Vector2d moveAnimal(Animal animal, Vector2d direction) {
```

Comment:
tu troochę skomplikował Pan zadanie, chcieliśmy tu użyć konkretnie listy, a nie tablicy. Pana rozwiązanie jest wydajniejsze, ale wymaga aktualizacji pozycji w mapie przy aktualizacji pozycji w zwierzątku - będziemy do tego tematu jeszcze wracać na dwóch kolejnych laborkach, ale na razie sugerowałbym to zwinąć do listy żeby łatwiej było zrobić dalsze rzeczy.